lib/pure/random.nim

```diff
@@ -196,37 +391,117 @@ proc sample*[T, U](r: var Rand; a: openArray[T], cdf: openArray[U]): T =
   a[cdf.upperBound(U(u))]
 
 proc sample*[T, U](a: openArray[T], cdf: openArray[U]): T =
-  ## Like ``sample(var Rand; openArray[T], openArray[U])``, but uses default
-  ## non-thread-safe state.
+  ## Returns an element from ``a`` using a cumulative distribution function
+  ## (CDF).
+  ##
+  ## This proc works similarly to
+  ## `sample[T, U](Rand, openArray[T], openArray[U])
+  ## <#sample,Rand,openArray[T],openArray[U]>`_.
+  ## See that proc's documentation for more details.
+  ##
+  ## If `randomize<#randomize,>`_ has not been called, the order of outcomes
+  ## from this proc will always be the same.
+  ##
+  ## This proc uses the default random number generator. Thus, it is **not**
+  ## thread-safe.
+  ##
+  ## See also:
+  ## * `sample proc<#sample,Rand,openArray[T],openArray[U]>`_ that also utilizes
+  ##   a CDF but uses a provided state
+  ## * `sample proc<#sample,openArray[T]>`_ that does not use a CDF
+  runnableExamples:
+    from math import cumsummed
+
+    let marbles = ["red", "blue", "green", "yellow", "purple"]
+    let count = [1, 6, 8, 3, 4]
+    let pick = sample(marbles, count.cumsummed)
   state.sample(a, cdf)
 
 proc initRand*(seed: int64): Rand =
-  ## Creates a new ``Rand`` state from ``seed``.
+  ## Initializes a new `Rand<#Rand>`_ state using the given seed.
+  ##
+  ## The resulting state is independent of the default random number
+  ## generator's state.
+  ##
+  ## Providing a specific seed will produce the same results for that
+  ## seed each time.
+  ##
+  ## See also:
+  ## * `randomize proc<#randomize,int64>`_ that accepts a seed for the default
+  ##   random number generator
+  ## * `randomize proc<#randomize,>`_ that initializes the default random
```

Comment:
These trailing commas in the links have been fixed in the last couple of days.
Use: `randomize proc<#randomize>`